components/nexusoperations/tasks.go

```diff
@@ -187,11 +199,17 @@ func (CancelationTask) Validate(ref *persistencespb.StateMachineRef, node *hsm.N
 type CancelationTaskSerializer struct{}
 
 func (CancelationTaskSerializer) Deserialize(data []byte, attrs hsm.TaskAttributes) (hsm.Task, error) {
-	return CancelationTask{EndpointName: attrs.Destination}, nil
+	attempt, _ := binary.Varint(data)
```

Comment:
It's fine to not error here since we may have tasks without any data attached but you can either make it more explicit by checking if `len(data) == 0` or at least add a comment explaining why you'd ignore an error here.

Reply:
`binary.Variant()` doesn't error, it returns the number of bytes read as the second value. So I think it is fine to leave it as the zero value. But irrelevant as I will replace with a struct